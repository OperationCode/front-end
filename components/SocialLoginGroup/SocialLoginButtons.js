import React from 'react';
import { func } from 'prop-types';
import classNames from 'classnames';
import GoogleLogin from 'react-google-login';
import FacebookLogin from 'react-facebook-login/dist/facebook-login-render-props';
import { clientTokens } from 'common/config/environment';
import styles from './SocialLoginGroup.css';

SocialLoginButtons.propTypes = {
  onSuccess: func.isRequired,
  onGoogleFailure: func.isRequired,
};

function SocialLoginButtons({ onSuccess, onGoogleFailure }) {
  return (
    <>
      <GoogleLogin
        clientId={clientTokens.OC_GOOGLE_KEY}
        buttonText="Login"
        onSuccess={onSuccess('google')}
        onFailure={onGoogleFailure}
        render={renderProps => (
          <button
            type="button"
            onClick={renderProps.onClick}
            disabled={renderProps.disabled}
            className={classNames(styles.loginButton, styles.googleButton)}
          >
            Login with Google
          </button>
        )}
      />

      <FacebookLogin
        appId={clientTokens.OC_FACEBOOK_KEY}
        callback={onSuccess('facebook')}
        redirectUri={typeof window === 'object' && `${window.location.origin}/login`}
        render={renderProps => (
          <button
            type="button"
            onClick={renderProps.onClick}
            className={classNames(styles.loginButton, styles.facebookButton)}
          >
            Login with Facebook
          </button>
        )}
      />
    </>
  );
}

export default SocialLoginButtons;
