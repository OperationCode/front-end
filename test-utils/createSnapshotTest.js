/* eslint-env jest */
import renderer from 'react-test-renderer';

/**
 * Used to create fully-rendered snapshot test.
 * Useful for reusable component smoke tests.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const tree = renderer.create(Component).toJSON();
  expect(tree).toMatchSnapshot();
};
