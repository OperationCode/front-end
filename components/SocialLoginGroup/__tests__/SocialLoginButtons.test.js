import React from 'react';
import createSnapshotTest from 'test-utils/createSnapshotTest';
import SocialLoginButtons from '../SocialLoginButtons';

describe('SocialLoginButtons', () => {
  beforeEach(() => {
    const fbScript = document.createElement('script');
    fbScript.id = 'facebook-jssdk';
    document.body.append(fbScript);
  });

  it('should render with required props', () => {
    createSnapshotTest(<SocialLoginButtons onGoogleFailure={jest.fn()} onSuccess={jest.fn()} />);
  });
});
