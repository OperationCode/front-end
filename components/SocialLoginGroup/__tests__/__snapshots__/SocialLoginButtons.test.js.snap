// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialLoginButtons should render with required props 1`] = `
Array [
  <button
    className="loginButton googleButton"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    Login with Google
  </button>,
  <button
    className="loginButton facebookButton"
    onClick={[Function]}
    type="button"
  >
    Login with Facebook
  </button>,
]
`;
